File: seed/php-model/trace/src/Problem/GetDefaultStarterFilesResponse.php
<?php

namespace Seed\Problem;

use Seed\Core\SerializableType;
use Seed\Commons\Language;
use Seed\Core\JsonProperty;
use Seed\Core\ArrayType;

class GetDefaultStarterFilesResponse extends SerializableType
{
    /**
     * @var array<Language, ProblemFiles> $files
     */
    #[JsonProperty("files"), ArrayType([Language::class => ProblemFiles::class])]
    public array $files;

    /**
     * @param array<Language, ProblemFiles> $files
     */
    public function __construct(
        array $files,
    ) {
        $this->files = $files;
    }
}

File: seed/pydantic-v2/trace/resources/problem/types/get_default_starter_files_response.py
from typing import Dict

from core.datetime_utils import serialize_datetime
from dt import datetime
from resources.commons.types.language import Language
from resources.problem.types.problem_files import ProblemFiles

from pydantic import BaseModel


class GetDefaultStarterFilesResponse(BaseModel):
    files: Dict[Language, ProblemFiles]

    class Config:
        frozen = True
        smart_union = True
        json_encoders = {datetime: serialize_datetime}
